### PR TITLE
Specs for Enumerator.new and yield arguments

### PR DESCRIPTION
**Pull request by andrykonchin**
Reproduce the issue described here https://github.com/oracle/truffleruby/issues/2016. 

It looks similar to a JRuby bug https://github.com/jruby/jruby/issues/4108 which is already fixed.

ping @eregon

### CODE REVIEW

**Comment by andrykonchin** [comment]
Looks like there is a problem with GitHub and Actions so will rerun specs later.

**Comment by andrykonchin** [comment]
Looks like one test is failing on Ruby 2.6 (but passes on all the other versions):

```ruby
Enumerator.new { |y| y.<<(1, 2) }.to_a.should == [[1, 2]]

# ArgumentError: wrong number of arguments (given 2, expected 1)
```

File: core/enumerator/new_spec.rb [hunk 1]
```diff
@@ -75,5 +75,33 @@ def each
...
+    end
+
+    it 'handles +yield+ arguments properly' do
+      Enumerator.new { |y| y.<<(1) }.to_a.should == [1]
+      Enumerator.new { |y| y.<<(1) }.first.should == 1
+
+      Enumerator.new { |y| y.<<([1]) }.to_a.should == [[1]]
+      Enumerator.new { |y| y.<<([1]) }.first.should == [1]
+
+      Enumerator.new { |y| y.<<(1, 2) }.to_a.should == [[1, 2]]
```
Comment (eregon) [inline core/enumerator/new_spec.rb:100]: It looks like this fails e.g. on 2.6

File: core/enumerator/new_spec.rb [hunk 2]
```diff
@@ -75,5 +75,33 @@ def each
...
+      Enumerator.new { |y| y.yield([1]) }.first.should == [1]
+
+      Enumerator.new { |y| y.yield(1, 2) }.to_a.should == [[1, 2]]
+      Enumerator.new { |y| y.yield(1, 2) }.first.should == [1, 2]
+
+      Enumerator.new { |y| y.yield([1, 2]) }.to_a.should == [[1, 2]]
+      Enumerator.new { |y| y.yield([1, 2]) }.first.should == [1, 2]
+    end
+
+    it 'handles +yield+ arguments properly' do
```
Comment (eregon) [inline core/enumerator/new_spec.rb:93]: Should include `<<` and be different than the previous spec description

**Comment by eregon** [comment]
> Looks like one new test is failing on Ruby 2.6 (but passes on all the other versions):

I would just not check that case, since it seems unlikely `<<` is used with multiple arguments.

**Comment by eregon** [comment]
> I would just not check that case, since it seems unlikely `<<` is used with multiple arguments.

Actually, even better would be to spec that it raises ArgumentError for Ruby >= 2.6

File: core/enumerator/yielder/append_spec.rb [hunk 3]
```diff
@@ -21,4 +21,27 @@
...
+        ary = []
+        y = Enumerator::Yielder.new { |*x| ary << x }
+        y.<<(1, 2)
+
+        ary.should == [[1, 2]]
+      end
+    end
+
+    ruby_version_is '2.6' do
+      it "yields the arguments list to the block" do
```
Comment (eregon) [inline core/enumerator/yielder/append_spec.rb:37]: `it "raises an ArgumentError" do`

Comment (eregon) [inline core/enumerator/yielder/append_spec.rb:37]:
```suggestion
      it "raises an ArgumentError" do
```

**Comment by eregon** [comment]
Thanks!

**Pull request merged**